src/main/java/ui/components/HTPanelMenuBar.java

```diff
@@ -0,0 +1,115 @@
+package ui.components;
+import backend.interfaces.IModel;
+import javafx.scene.control.Label;
+import ui.issuepanel.FilterPanel;
+import javafx.application.Platform;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.HBox;
+
+/**
+ * An HTPanelMenuBar allows for the creation of a high level panel menu bar interface that
+ * contains logic for renaming the panel menu, handling the button and keyboard events
+ * and augmenting the children of panel menu bar defined in FilterPanel.java during the
+ * renaming process. The class is instantiated in FilterPanel.java.
+ */
+public class HTPanelMenuBar extends HBox {
```

Comment:
@dariusf do we use `Hub` as the prefix or `Ht`? Use whatever we use in other places.